### ISSUE TITLE
Fix Ruby 2.x.x keyword warnings when using argument forwarding

### ISSUE THREAD

**Issue by bkuhlmann** [dry-rb/dry-schema#449]
## Describe the bug

Hello. :wave: I'm seeing the following warnings when using Dry Schema with [Ruby 3.2.0 RC1](https://www.ruby-lang.org/en/news/2022/12/06/ruby-3-2-0-rc1-released):

``` 
lib/dry/schema/macros/dsl.rb:181: warning: Skipping set of ruby2_keywords flag for array (method accepts keywords or method does not accept argument splat)
```

## To Reproduce

The quickest way to reproduce is to install [Ruby 3.2.0 RC1](https://www.ruby-lang.org/en/news/2022/12/06/ruby-3-2-0-rc1-released) using your favorite [Ruby Version Manager](https://www.alchemists.io/articles/ruby_version_managers) or you can use my *latest* Docker [image](https://www.alchemists.io/projects/docker-alpine-ruby) which is running Ruby 3.2.0 with YJIT. Then you can run the following code snippet: 

``` ruby
#! /usr/bin/env ruby
# frozen_string_literal: true

# Save as `snippet`, then `chmod 755 snippet`, and run as `./snippet`.

require "bundler/inline"

gemfile true do
  source "https://rubygems.org"
  gem "dry-schema"
end

Demo = Dry::Schema.Params { required(:demo).filled :string }
```

## Expected behavior

I'd like to not see the warnings being printed. To fix -- at least when tinkering locally -- I deleted `ruby2_keywords` lines from the following files to fix the problem:

``` 
lib/dry/schema/macros/dsl.rb
lib/dry/schema/macros/key.rb
```

Because Dry Schema is using *argument forwarding*, it would be simple enough to delete use of `ruby2_keywords` if you are willing to release a version that supports Ruby 3.2.0 or higher. It might be possible to use a minimum version of Ruby 3.1.x but not sure (have not tested on 3.1 or 3.0).

## My environment

- Affects my production application: **NO**
- Ruby version: ruby 3.2.0rc1 (2022-12-18) [arm64-darwin22.2.0]
- OS: macOS 13.1
- Dry Schema: 1.13.0

**Comment by esparta** [comment]
The issue was fixed on #447 and will be soon on the next release, @bkuhlmann .

**Comment by bkuhlmann** [comment]
Hey Espartaco, thanks! I was searching through issues and didn't see this mentioned but didn't search through closed pull requests. Good to know and looking forward to the next release. :bow:
